src/main/java/org/spongepowered/api/schematic/Schematic.java

```diff
@@ -0,0 +1,210 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.schematic;
+
+import com.flowpowered.math.vector.Vector3i;
+import com.google.common.collect.BiMap;
+import org.spongepowered.api.block.BlockSnapshot;
+import org.spongepowered.api.block.BlockState;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataSerializable;
+import org.spongepowered.api.data.persistence.DataBuilder;
+import org.spongepowered.api.data.persistence.InvalidDataException;
+import org.spongepowered.api.world.World;
+import org.spongepowered.api.world.extent.Extent;
+
+import java.util.Optional;
+
+/**
+ * Unless otherwise noted, "the specification" refers to the latest version of the Schematic Specification {@url https://github
+ * .com/SpongePowered/Schematic-Specification}.
+ */
+public interface Schematic extends DataSerializable {
+
+    /**
+     * Gets the attached {@link DataContainer}.
+     *
+     * <p>Correlates with the optional `Metadata` field of the specification.</p>
+     *
+     * @return The attached {@link DataContainer} or {@link Optional#empty()} if none
+     */
+    Optional<DataContainer> getMetadata();
+
+    /**
+     * Gets the width of the Schematic.
+     *
+     * <p>Correlates with the `Width` field of the specification.</p>
+     *
+     * @return The width of the Schematic.
+     */
+    int getWidth();
+
+    /**
+     * Gets the height of the Schematic.
+     *
+     * <p>Correlates with the `Height` field of the specification.</p>
+     *
+     * @return The height of the Schematic.
+     */
+    int getHeight();
+
+    /**
+     * Gets the length of the Schematic.
+     *
+     * <p>Correlates with the `Length` field of the specification.</p>
+     *
+     * @return The length of the Schematic.
+     */
+    int getLength();
+
+    /**
+     * Gets the origin of the Schematic (relative to the minimum point) or the minimum point if none.
+     *
+     * <p>Correlates with the `OffsetX`, `OffsetY`, and `OffsetZ` fields of the specification.</p>
+     *
+     * @return The origin of the Schematic
+     */
+    Vector3i getOrigin();
+
+    /**
+     * Gets the largest key in the palette.
+     *
+     * @return The largest key
+     */
+    int getPaletteMax();
```

Comment:
It doesn’t. The palette could have spots, so size isn’t the biggest key. You have to take the Set of keys and sort them.
> On Apr 4, 2016, at 1:51 PM, Aaron Hill <notifications@github.com> wrote:
> 
> In src/main/java/org/spongepowered/api/schematic/Schematic.java <https://github.com/SpongePowered/SpongeAPI/pull/1172#discussion_r58427606>:
> 
> > +
> > +    /**
> > +     * Gets the origin of the Schematic (relative to the minimum point) or the minimum point if none.
> > +     *
> > +     * <p>Correlates with the `OffsetX`, `OffsetY`, and `OffsetZ` fields of the specification.</p>
> > +     *
> > +     * @return The origin of the Schematic
> > +     */
> > +    Vector3i getOrigin();
> > +
> > +    /**
> > +     * Gets the largest key in the palette.
> > +     *
> > +     * @return The largest key
> > +     */
> > +    int getPaletteMax();
> I know that this is in the specification, but it's more of an implementation detail. getPalette#size already does this.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/SpongePowered/SpongeAPI/pull/1172/files/3e69789b55ffb54141cfed6d82c3b58d07c707f5#r58427606>